### ISSUE TITLE
Operator zap option flags ignored

### ISSUE THREAD

**Issue by brusdev** [artemiscloud/activemq-artemis-operator#643]
The zap option flags defined by the command args and the environment variable `ARGS` are ignored. See the following deployment and logs:

```
    spec:
      containers:
        - name: manager
          image: quay.io/artemiscloud/activemq-artemis-operator:1.0.13
          command:
            - /home/activemq-artemis-operator/bin/entrypoint
          args:
            - '--zap-log-level=info'
            - '--zap-encoder=console'
            - '--zap-time-encoding=iso8601'
            - '--leader-elect'
```

```
Executing entrypoint
exec /home/activemq-artemis-operator/bin/activemq-artemis-operator
{"level":"info","ts":"2023-07-10T08:01:21Z","logger":"setup","msg":"Go Version: go1.19.10"}
{"level":"info","ts":"2023-07-10T08:01:21Z","logger":"setup","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":"2023-07-10T08:01:21Z","logger":"setup","msg":"Version of operator-sdk: 1.28.0"}
{"level":"info","ts":"2023-07-10T08:01:21Z","logger":"setup","msg":"Version of the operator: 1.0.13 2023-06-29T04:50:19"}
{"level":"info","ts":"2023-07-10T08:01:21Z","logger":"setup","msg":"Supported ActiveMQArtemis Kubernetes Image Versions: 2.21.0 2.22.0 2.23.0 2.25.0 2.26.0 2.27.0 2.27.1 2.28.0 2.29.0"}
{"level":"info","ts":"2023-07-10T08:01:21Z","logger":"setup","msg":"Got operator namespace","operator ns":"default"}
{"level":"info","ts":"2023-07-10T08:01:21Z","logger":"setup","msg":"setting up operator to watch local namespace"}
{"level":"info","ts":"2023-07-10T08:01:21Z","logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":"0.0.0.0:8383"}
{"level":"info","ts":"2023-07-10T08:01:22Z","logger":"setup","msg":"service account name: activemq-artemis-controller-manager"}
{"level":"info","ts":"2023-07-10T08:01:22Z","logger":"setup","msg":"NOT Setting up webhook functions","ENABLE_WEBHOOKS":"false"}
{"level":"info","ts":"2023-07-10T08:01:22Z","logger":"setup","msg":"starting the Cmd."}
{"level":"info","ts":"2023-07-10T08:01:22Z","logger":"controller_v1beta1activemqartemisaddress","msg":"Setting up address observer"}
{"level":"info","ts":"2023-07-10T08:01:22Z","msg":"Starting server","path":"/metrics","kind":"metrics","addr":"[::]:8383"}
{"level":"info","ts":"2023-07-10T08:01:22Z","msg":"Starting server","kind":"health probe","addr":"[::]:8081"}
{"level":"info","ts":"2023-07-10T08:01:22Z","logger":"addressobserver_activemqartemisaddress","msg":"#### Started workers"}
```

**Comment by brusdev** [comment]
This issue was caused by #619 that updated github.com/RHsyseng/operator-utils to v1.4.12. Since operator-utils v1.4.11 the compare module calls the function [SetLogger](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.1/pkg/log#SetLogger) setting its logger, see [[RHPAM-4438] Sensitive Information In ENV And Pod Logs](https://github.com/RHsyseng/operator-utils/commit/7a9086dae039b0dc22aaa8a64462f036bdc770a1). This change prevents operator main to set its logger.

**Comment by brusdev** [comment]
This issue was fixed reverting #619
